extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/binder/MongoParserVisitor.java

```diff
@@ -54,7 +54,13 @@ public String visitLessThanOrEqualPredicate(HqlParser.LessThanOrEqualPredicateCo
 
     @Override
     public String visitLikePredicate(HqlParser.LikePredicateContext ctx) {
-        return ctx.expression(0).accept(this) + ":{'$regex':" + ctx.expression(1).accept(this) + "}";
+        String parameter = ctx.expression(1).accept(this);
+        if (parameter.indexOf('/') == 1 && parameter.lastIndexOf('/') > 1) {
```

Comment:
what's in position 0 here?  i would expect that to be the leading /

Reply:
This means that my comment is not clear enought ;)
`ctx.expression(1).accept(this);` will calls `visitParameterExpression()` that will escape the parameter with single quote.
But if the parameter is a JavaScript patern, we need to un-escape it as the character in position.

So the parameter in position zero is `'`.

I didn't want to add regex specific stuff inside `visitParameterExpression()` ... mainly because there is no easy way to know inside this method that we are in a like expression (maybe possible via ctx.getParent() ...)

Reply:
Ah, ok.  I wasn't sure if the single quote there was part of the incoming string or a typographical artifact to delinate a string a text.  :)